surveys/2022-wg-debugging-survey/questions.md

```diff
@@ -0,0 +1,360 @@
+Note: this is specifically about which systems you use for development and
+debugging not all the systems you target.
+
+Type: select all that apply (optional)
+
+- Linux
+- Windows
+- Windows Subsystem for Linux
+- Mac OS
+- Embedded devices
+- Other (open response)
+
+> *Justification:* Cohort analysis
+>
+> Debuggers are often platform-specific and various improvements that the
+> compiler team could make could only benefit a subset of all platforms, so
+> knowing what users are using is valuable.
+
+### How would you rate your Rust expertise?
+Type: select one
+
+- I can't read or write Rust
+- I can write simple programs in Rust
+- I can write useful, production-ready code but it is a struggle
+- I am productive writing Rust
+- I am an expert
+
+> *Justification:* Cohort analysis
+>
+> Users with different levels of experience are likely to face different
+> challenges.
+
+
+### What forms of debugging do you typically use with Rust?
+Type: select all that apply
+
+- `println!`/`log`/`tracing`
+- Stack traces
+- Traditional debuggers
+- Record/replay debuggers
+- High-level tracing (`dtrace`/`ebpf`/`ETW`)
+
+> *Justification:*
+>
+> Understand how users are debugging Rust code.
+
+### What uses of Rust are you debugging?
+Type: select all that apply (optional)
+
+- Single-threaded application
+- Multi-threaded application
+- Pure Rust application
+- Mixed C or C++ and Rust application
+- Crash dumps
+- Cloud/server-side application
+- Embedded hardware
+
+> *Justification:*
+>
+> Some debugger use cases are very different from "standard" debugging.
+
+### Do you debug Rust remotely?
+i.e. are you debugging something that's running on an embedded device/another
+platform, or something like that
+
+Type: select all that apply
+
+- Yes, debugger on the host (e.g. `gdbserver` or wired connection to embedded device)
+- Yes, debugger on the device (e.g. an SSH session with a debugger running)
+- No
+
+> *Justification:*
+>
+> Remote debugging will have a different user experience that might be better
+> or worse.
+
+### When do you start debugging Rust?
+Type: select all that apply
+
+- Fixing a crash/panic/access violation
+- Fixing a logic bug
+- Learning how software works
+- Understand runtime semantics
+- Understand a third-party library's behaviour
+- Understand own program's behaviour
+- Profiling
+- Investigating a crash dump
+
+> *Justification:*
+>
+> Different motivations for debugging will benefit from different features from
+> debuggers and might have a better or worse experience.
+
+
+### How would you rate your expertise with debuggers?
+Type: select one
+
+- I have never used a debugger
+- I am uncomfortable using a debugger
+- I am comfortable using a debugger
+- I am very comfortable using a debugger
+
+> *Justification:*
+>
+> Users who are more familiar with debuggers will face different problems than
+> users who are not.
+
+### Why are you uncomfortable using a debugger?
```

Comment:
We should also be explicit here about this being about using a debugger "with Rust"

Reply:
I haven't added that because the question that it follows-on from isn't just about "with Rust", and I think that we might want this question to be more general (I think it could be worth having a "in general" and "with Rust specifically" question for this)